clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -719,38 +726,66 @@ public Node provide() {
 
     abstract class Call {
         private final boolean internal;
+        private final boolean supportsUseControllers;
         private final String callName;
         private final long deadlineMs;
         private final NodeProvider nodeProvider;
         protected int tries;
         private Node curNode = null;
         private long nextAllowedTryMs;
 
-        Call(boolean internal,
-             String callName,
-             long nextAllowedTryMs,
-             int tries,
-             long deadlineMs,
-             NodeProvider nodeProvider
+        Call(
```

Comment:
The other constructors are all supplying different defaults to `supportsUseControllers`. Is that expected? When should a constructor default true vs false?

Reply:
I agree that this is a bit confusing. I'll cut it down to just 3 constructors.